### PR TITLE
Add `pip install[X]` for all deps and document

### PR DESCRIPTION
**Pull request by m-col**

### CODE REVIEW

**Comment by elParaguayo** [comment]
Can all the stuff in `setup.cfg` be moved to `pyproject.toml`?

**Comment by elParaguayo** [comment]
Oh, and great idea too!

**Comment by m-col** [comment]
> Can all the stuff in setup.cfg be moved to pyproject.toml?

Most of it I think so yeah :) one thing at a time!

File: setup.cfg [hunk 1]
```diff
@@ -39,6 +39,27 @@ wayland =
...
+  pytz
+  xdg
+  xmltodict
+all =
+  # All user dependencies i.e. excluding dev deps.
+  %(ipython)s
+  %(wayland)s
+  %(widgets)s
+  bowler
+  dbus-next
```
Comment (elParaguayo) [inline setup.cfg:61]: This is included in `widgets`

Comment (m-col) [reply setup.cfg:61]: Ah nice

Comment (m-col) [reply setup.cfg:61]: Ooh the widgets one has an underscore and the test one has a dash. I guess pip normalises those when installing things

File: docs/manual/install/index.rst [hunk 2]
```diff
@@ -116,6 +116,13 @@ With the dependencies in place, you can now install the stable version of qtile
 
    pip install qtile
 
+Or with sets of dependencies:
+
+.. code-block:: bash
+
+   pip install qtile[wayland]  # for Wayland dependencies
+   pip install qtile[widgets]  # for all widget dependencies
+   pip install qtile[all]      # for all user dependencies
```
Comment (jwijenbergh) [inline docs/manual/install/index.rst:125]: Great idea!

From an outside view maybe "user" dependencies is a bit vague. Is this documented somewhere?

Comment (jwijenbergh) [reply docs/manual/install/index.rst:125]: If it's not documented a small sentence somewhere above or in the comment itself will do

Comment (m-col) [reply docs/manual/install/index.rst:125]: Good point, maybe just "all dependencies"?

File: setup.cfg [hunk 3]
```diff
@@ -39,6 +39,26 @@ wayland =
...
   pywlroots==0.16.4
+widgets =
+  dbus-next
+  imaplib2
+  iwlib
+  keyring
+  mailbox
+  psutil
+  pulsectl
+  pulsectl_asyncio
```
Comment (elParaguayo) [inline setup.cfg:50]: I think `pulsectl_asyncio` pulls in `pulsectl`

File: docs/manual/install/index.rst (hunk 2 above)
Comment (jwijenbergh) [reply docs/manual/install/index.rst]: Oh that is much less confusing

File: setup.cfg (hunk 3 above)
Comment (jwijenbergh) [reply setup.cfg:50]: It does indeed

Comment (m-col) [reply setup.cfg:50]: We import pulsectl so figured it is a dep in its own right. Im not so familiar with the widget or those libs though so maybe they are always together?

**Pull request merged**